syncedlyrics/__init__.py

```diff
@@ -11,29 +11,42 @@
 from typing import List, Optional
 
 from .providers import Deezer, Lrclib, Musixmatch, NetEase, Megalobiz, Genius
-from .utils import is_lrc_valid, save_lrc_file
+from .utils import Lyrics, TargetType
+from .providers.base import LRCProvider
 
 logger = logging.getLogger(__name__)
 
 
 def search(
     search_term: str,
     allow_plain_format: bool = False,
```

Comment:
I think introducing a class as the response of the `search` function will break the compatibility anyway. So allow plain would be no longer necessary instead we can replace it with `plaintext_only`.

Reply:
Well yeah, I was unsure how you feel about it, I also think it would be cleaner to return the class, rename the "allow_plaintext" to "plaintext_only", even though it breaks compatibility.
However in the current form I tried not to break compatibility and therefore made the search function still return a string (selecting from the Lyrics class according to the preference). If you want I can quickly change that.

Reply:
Sorry I thought you returned the `Lyrics` instance . In that case I think string is simpler and doesn't break anything so all good no change needed.